drivers/timer/hi3861_timer.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT hisilicon_hi3861_timer
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/irq.h>
+#include <zephyr/arch/common/sys_io.h>
+
+#define TIMER_REG DT_INST_REG_ADDR(0)
+
+#define TIMER_0 (TIMER_REG + 0x00)
+#define TIMER_1 (TIMER_REG + 0x14)
+#define TIMER_2 (TIMER_REG + 0x28)
+#define TIMER_3 (TIMER_REG + 0x3C)
+
+#define TIMER_LOADCOUNT_REG(BASE)    (BASE + 0x000)
+#define TIMER_CURRENTVALUE_REG(BASE) (BASE + 0x004)
+#define TIMER_CONTROLREG_REG(BASE)   (BASE + 0x008)
+#define TIMER_CONTROLREG_EN	     (1 << 0)
+#define TIMER_CONTROLREG_MODE_FREE   (0 << 1)
+#define TIMER_CONTROLREG_MODE_CYCLE  (1 << 1)
+#define TIMER_CONTROLREG_INT_MASK    (1 << 2)
+#define TIMER_CONTROLREG_INT_UNMASK  (0 << 2)
+#define TIMER_CONTROLREG_LOCK	     (1 << 3)
+#define TIMER_EOI_REG(BASE)	     (BASE + 0x00C)
+#define TIMER_INTSTATUS_REG(BASE)    (BASE + 0x010)
+
+#define SYSTICK TIMER_3
+
+#define CYCLES_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static volatile uint64_t last_ticks;
+
+static void sys_clock_isr(const void *arg)
+{
+	ARG_UNUSED(arg);
+
+	/* Read EOI to clear interrupt */
+	sys_read32(TIMER_EOI_REG(SYSTICK));
+
+	last_ticks++;
+
+	sys_clock_announce(1);
```

Comment:
I'm still very much curious why you are going with a tickless kernel. If this is a regular counter, why not using a proper tickfull implementation?

Reply:
No, it's not tickless. The timer, working at 24MHz, is set to announce `1` tick every `240000` cycles.

Reply:
you mean that you cannot set a timeout value for the timer?

Reply:
It can. But I didn't implement the tickless feature. It is required?

Reply:
Thank you for your suggestion. It seems that I have confused the meanings of "tickless" and "tickfull". I will update later.